src/expr.rs

```diff
@@ -688,6 +688,13 @@ fn rewrite_closure_expr(
     if classify::expr_requires_semi_to_be_stmt(left_most_sub_expr(expr)) {
         rewrite = and_one_line(rewrite);
     }
+    rewrite = rewrite.and_then(|rw| {
+        if !context.config.overflow_closures() && rw.contains('\n') {
```

Comment:
(Totally not a `rustfmt` expert, so could totally be wrong, but won't this undo any other formatting on everything else inside the expression?)